### PR TITLE
FilesTimes support does not build for ESP-IDF

### PR DESCRIPTION
**Pull request by ivmarkov**
Commit https://github.com/rust-lang/rust/commit/1f5d8d49eb6111931091f700d07518cd2b80bc18 broke STD for `target_os = "espidf"`. 

In future, we might come up with something more sophisticated (as in using the `utime` function which *is* available on the ESP-IDF platform), but for now we are treating ESP-IDF just like `Redox` in that the new API fails at runtime. Most important for us ATM is to restore successful compilation of STD on our platform.

### CODE REVIEW

File: library/std/src/sys/unix/fs.rs [hunk 1]
```diff
@@ -1077,8 +1079,10 @@ impl File {
 
     pub fn set_times(&self, times: FileTimes) -> io::Result<()> {
         cfg_if::cfg_if! {
-            if #[cfg(target_os = "redox")] {
+            if #[cfg(any(target_os = "redox", target_os = "espidf"))] {
                 // Redox doesn't appear to support `UTIME_OMIT`.
+                // ESP-IDF does not support the `futimens` function.
+                // It would need a `utime`-based code-path in future.
```
Comment (joshtriplett) [inline library/std/src/sys/unix/fs.rs:1085]:
```suggestion
```

`utime` is filename-based, not fd-based. Does ESP-IDF have an fd-based API to use here?

In any case, let's not speculate on what path future additions might take here.

**Comment by joshtriplett** [comment]
@ivmarkov r=me with the suggestion applied.

File: library/std/src/sys/unix/fs.rs (hunk 1 above)
Comment (ivmarkov) [reply library/std/src/sys/unix/fs.rs]: Agreed and comment fixed accordingly.

**Pull request merged**